tests/unit/test_collector.py

```diff
@@ -31,3 +31,126 @@ async def test_get_stats(self, collector_daemon):
         """Test get_stats function and the execution of the collector."""
         statsd = collector_daemon()
         await statsd.get_stats()
+        assert statsd.currently_cached_labels == {
+            "juju-000ddd-test-0": {
+                "job": "prometheus-juju-exporter",
+                "hostname": "juju-000ddd-test-0",
+                "customer": "example_customer",
+                "cloud_name": "example_cloud",
+                "juju_model": "test",
+                "type": "kvm",
```

Comment:
MachineType should be StrEnum

Reply:
Since `StrEnum` only made available to `enum` in Python 3.11, I'm using a simple class to simulate `enum` as we discussed.